src/v2/Utils/openAuthModal.ts

```diff
@@ -13,24 +13,28 @@ export interface AuthModalOptions extends ModalOptions {
 }
 
 export const openAuthModal = (mediator: Mediator, options: ModalOptions) => {
-  if (authModalReady(mediator)) {
-    mediator.trigger("open:auth", options)
+  triggerEvent(mediator, "open:auth", options)
+}
+
+export const triggerEvent = (
+  mediator: Mediator,
+  eventName: string,
+  options?: ModalOptions
+) => {
+  if (mediator.ready(eventName)) {
+    mediator.trigger(eventName, options)
     return
   }
 
   const intervalId = setInterval(() => {
-    if (authModalReady(mediator)) {
-      mediator.trigger("open:auth", options)
+    if (mediator.ready(eventName)) {
+      mediator.trigger(eventName, options)
       clearInterval(intervalId)
     }
   }, 100)
 }
 
-const authModalReady = (mediator: Mediator): boolean => {
-  return mediator.ready("open:auth")
-}
-
-export const openAuthToFollowSave = (
+export const openAuthToFollowSaveCreate = (
```

Comment:
**question**: I'm struggling to understand this function. Does this handle the open auth behavior for following, saving, and creating alerts? Is there a way to extract the create alert tracking or maybe we just use `openAuthModal` directly?

Reply:
It handles the open auth behavior for following artist/partner/gene, saving artwork, and creating an alert by customizing options for follow up actions depending on passed intent. We can pass necessary options directly to `openAuthModal`, but I think this logic is similar to two others, + it handles redirecting user to _/login_ in case opening auth modal is not handled properly for some reason.

What do you think of this?

Reply:
Maybe in that case we can rename the function to something more generic like `openAuthToSatisfyIntent` if the function is able to figure out which intent is active and respond accordingly?

Reply:
Good idea!